.github/workflows/call-windows-unit-tests.yaml

```diff
@@ -0,0 +1,180 @@
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: ${{ matrix.config.arch }}
+
+      - name: Get gzip command w/ chocolatey
+        uses: crazy-max/ghaction-chocolatey@v3
+        with:
+          args: install gzip -y
+
+      # http://man7.org/linux/man-pages/man1/date.1.html
+      - name: Get Date
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Restore cached packages of vcpkg
+        id: cache-unit-test-vcpkg-sources
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            C:\vcpkg\packages
+          key: ${{ runner.os }}-${{ matrix.config.arch }}-wintest-vcpkg-${{ steps.get-date.outputs.date }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.config.arch }}-wintest-vcpkg-
+          enableCrossOsArchive: false
+
+      - name: Build openssl with vcpkg
+        run: |
+          C:\vcpkg\vcpkg install --recurse openssl --triplet ${{ matrix.config.vcpkg_triplet }}
+        shell: cmd
+
+      - name: Build libyaml with vcpkg
+        run: |
+          C:\vcpkg\vcpkg install --recurse libyaml --triplet ${{ matrix.config.vcpkg_triplet }}
+        shell: cmd
+
+      - name: Save packages of vcpkg
+        id: save-vcpkg-sources
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            C:\vcpkg\packages
+          key: ${{ steps.cache-unit-test-vcpkg-sources.outputs.cache-primary-key }}
+          enableCrossOsArchive: false
+
+      - name: run unit-test for Fluent Bit packages (only for x86 and x64)
+        run: |
+          cmake -G "NMake Makefiles" `
+            -D FLB_TESTS_INTERNAL=On `
+            -D FLB_NIGHTLY_BUILD='${{ inputs.unstable }}' `
+            -D OPENSSL_ROOT_DIR='${{ matrix.config.openssl_dir }}' `
+            ${{ matrix.config.cmake_additional_opt }} `
+            -D FLB_LIBYAML_DIR=C:\vcpkg\packages\libyaml_${{ matrix.config.vcpkg_triplet }} `
+            -D FLB_WITHOUT_flb-rt-out_elasticsearch=On `
+            -D FLB_WITHOUT_flb-rt-out_td=On `
+            -D FLB_WITHOUT_flb-rt-out_forward=On `
+            -D FLB_WITHOUT_flb-rt-in_disk=On `
+            -D FLB_WITHOUT_flb-rt-in_proc=On `
+            -D FLB_WITHOUT_flb-it-parser=On `
+            -D FLB_WITHOUT_flb-it-unit_sizes=On `
+            -D FLB_WITHOUT_flb-it-network=On `
+            -D FLB_WITHOUT_flb-it-pack=On `
+            -D FLB_WITHOUT_flb-it-signv4=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_ec2=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_http=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_profile=On `
+            -D FLB_WITHOUT_flb-it-aws_credentials_sts=On `
+            -D FLB_WITHOUT_flb-it-aws_util=On `
+            -D FLB_WITHOUT_flb-it-input_chunk=On `
+            ../
+          cmake --build .
+          ctest --build-run-dir $PWD --output-on-failure
```

Comment:
Ah, I suggest we split these two lines into separate jobs to run unit tests ephemerally and/or upload the artefacts first.

You can split it into separate steps with an upload in between then people can download them if the tests fail to try locally or otherwise look at.

If we upload the binaries we can then have a separate job that downloads them and runs the tests in a clean runner.

Reply:
This workflow is written to be referred from the previous step using workflow which defines pr-windows-build job...

Reply:
Sorry, I mean upload the test binaries - these are never built currently.

Reply:
Yeah I know, however it would catch any new ones we might have missed or whatever. For not though it might be overkill.

We could also add a step just to dumpbin the dlls as well.

Reply:
I added a step to dump dependent DLLs.
https://github.com/fluent/fluent-bit/pull/8582/commits/c0ce01ff4a74f0b34f333e42a7f24f6ad02b0732

However, this step is just dumping the DLL dependencies.

Reply:
Also, we could mark as a failure when the not system linkages are detected but I supposed that it should be overkill for now.